libsrc/common/moveinit.s

```diff
@@ -0,0 +1,54 @@
+;
+; 2015-10-04, Greg King
+;
+
+        .export         move_init
+
+        .import         __INIT_LOAD__, __INIT_RUN__, __INIT_SIZE__ ; Linker-generated
+        .importzp       init_load_, init_run_
+
+        .macpack        cpu
+        .macpack        generic
+
+
+; Move the INIT segment from where it was loaded (over the bss segments)
+; into where it must be run (in the heap).  The two areas might overlap; and,
+; the segment is moved upwards.  Therefore, this code starts at the highest
+; address, and decrements to the lowest address.  The low bytes of the starting
+; pointers are not sums.  The high bytes are sums; but, they do not include the
+; carry.  Both the low-byte sums and the carries will be done when the pointers
+; are indexed by the .Y register.
+
+move_init:
+        lda     #<__INIT_LOAD__
+        ldx     #>__INIT_LOAD__ + >__INIT_SIZE__
+        sta     init_load_
+        stx     init_load_+1
+        lda     #<__INIT_RUN__
+        ldx     #>__INIT_RUN__ + >__INIT_SIZE__
+        sta     init_run_
+        stx     init_run_+1
```

Comment:
My partial implementation of the move routine looks (of course) pretty much like yours. Howeve I thought that it is used exactly once and therefore it might be appropriate to put it in the DATA segment and use selfmodifying code instead of indirect adressing in order to save space. This approach would basically remove the code block above...